be/src/column/hash_set.h

```diff
@@ -80,4 +80,42 @@ struct PhSetTraits<Slice> {
 template <typename T>
 using PhSet = typename PhSetTraits<T>::SetType;
 
+struct SliceKey8 {
+    union U {
+        struct {
+            char data[7];
+            uint8_t size;
```

Comment:
is this padding safe? will some compiler treat it like this
```
char data[7];
char pad1[1];
uint8_t size;
char pad2[7];

Reply:
Not sure. I'd better add some static_assert here or use the packed attribute.